terracotta/drivers/mysql.py

```diff
@@ -354,10 +355,19 @@ def get_datasets(self, where: Mapping[str, str] = None,
             if not all(key in self.key_names for key in where.keys()):
                 raise exceptions.InvalidKeyError('Encountered unrecognized keys in '
                                                  'where clause')
-            where_fragment = ' AND '.join([f'{key}=%s' for key in where.keys()])
+            where = {
+                key: value if isinstance(value, list) else [value]
+                for key, value in where.items()
+            }
+            conditions = [
+                '(%s)' % ' OR '.join([f'{key}=%s'] * len(value))
+                for key, value in where.items()
+            ]
+            values = list(itertools.chain(*where.values()))
+            where_fragment = ' AND '.join(conditions)
```

Comment:
This is going a bit too far with the comprehensions. I think a straightforward loop might be more readable?